### PR TITLE
UM Hackathon 2021: Add 'Ribbed Support Vaults' infill pattern.

### PR DESCRIPTION
**Pull request by rburema**
This just adds the pattern as an option.

W.I.P.! See engine (draft) PR for progress, as it is the main thing worked on: https://github.com/Ultimaker/CuraEngine/pull/1395

Tagged Uranium with a branch as well to have a relatively stable dev environment: https://github.com/Ultimaker/Uranium/tree/UMH-2021_ribbed_vaults_infill

### CODE REVIEW

**Comment by fieldOfView** [comment]
Aww, you had a hackathon?

File: resources/definitions/fdmprinter.def.json [hunk 1]
```diff
@@ -1748,7 +1748,7 @@
                 "zig_zaggify_infill":
                 {
                     "label": "Connect Infill Lines",
-                    "description": "Connect the ends where the infill pattern meets the inner wall using a line which follows the shape of the inner wall. Enabling this setting can make the infill adhere to the walls better and reduce the effects of infill on the quality of vertical surfaces. Disabling this setting reduces the amount of material used.",
+                    "description": "Connect the ends where the infill pattern meets the inner wall using a line which follows the shape of the inner wall. Enabling this setting can make the infill adhere to the wallevs better and reduce the effects of infill on the quality of vertical surfaces. Disabling this setting reduces the amount of material used.",
```
Comment (fieldOfView) [inline resources/definitions/fdmprinter.def.json:1751]: This looks like an unintended change/a typo.

**Comment by nallath** [comment]
> Aww, you had a hackathon?

Still ongoing (6 afternoons, we've had 3 now) It's not quite the format that we want it to have because of Corona, but I'm pretty pleased that it's happening. It's even across the whole of R&D (instead of just SW as the previous ones were).

File: resources/definitions/fdmprinter.def.json (hunk 1 above)
Comment (rburema) [reply resources/definitions/fdmprinter.def.json:1751]: Oops!

**Comment by rburema** [comment]
@fieldOfView  I somehow missed your message. Yes we do have a Hackathon! Tomorrow afternoon, and the day after and next week Wednesday still remaining.

**Review by jellespijker — APPROVED** [review]
See comment on the CuraEngine PR.

Still needs to go through component testing

**Comment by Ghostkeeper** [comment]
This one could be great as a support pattern too by the way, when used with an extra support wall. Could be very nice to combine with tree support to finally have a satisfactory solution for [this issue](https://github.com/Ultimaker/Cura/issues/4342).

**Comment by jellespijker** [comment]
I think you guys made an excellent addition to Cura. This infill generator uses the bare minimum on material, while still overing support for top skin layers. It has some interesting use cases.
lightning

**Pull request merged**